ocs_ci/utility/pagerduty.py

```diff
@@ -45,6 +47,41 @@ def set_pagerduty_integration_secret(integration_key):
     logger.info("New integration key was set.")
 
 
+def set_pagerduty_faas_secret(integration_key):
+    """
+    Update managed-fusion-agent-config secret. This is valid only on Fusion aaS platform.
+    managed-fusion-agent-config secret is expected to be present prior to the update.
+
+    Args:
+        integration_key (str): Integration key taken from PagerDuty Prometheus integration
+
+    """
+    logger.info("Setting up PagerDuty")
+    kubeconfig = os.getenv("KUBECONFIG")
+    ns_name = config.ENV_DATA["service_namespace"]
+    pd_configuration = (
+        f"    sopEndpoint: <pager duty service endpoint>\n"
+        f"    serviceKey: {integration_key}"
+    )
+    pd_configuration = pd_configuration.encode("utf-8")
+    pd_configuration = base64.b64encode(pd_configuration)
+    pd_configuration = pd_configuration.decode("utf-8")
+    cmd = f"oc get secret {constants.FUSION_AGENT_CONFIG_SECRET} -n {ns_name} -o yaml"
+    secret_data = exec_cmd(cmd).stdout
+    secret_data = yaml.safe_load(secret_data)
+    secret_data["data"]["pager_duty_config"] = pd_configuration
+    secret_data["metadata"].pop("resourceVersion", None)
+    secret_data["metadata"].pop("uid", None)
+
+    with tempfile.NamedTemporaryFile(
+        mode="w+", prefix=f"{constants.FUSION_AGENT_CONFIG_SECRET}_"
+    ) as secret_file:
+        yaml.dump(secret_data, secret_file)
+        secret_file.flush()
+        exec_cmd(f"oc apply --kubeconfig {kubeconfig} -f {secret_file.name}")
+    logger.info("New PagerDuty service was set.")
```

Comment:
It could be helpful to add a comment or a log message that describes the process here.

Reply:
The log was expanded.